src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+            {
+                tile.getMicroblockContainer().getPartContainer().onClicked(player, (RayTraceResult) hit);
+            }
+        }
+        else
+        {
+            onBlockClickedDefault(world, pos, player);
+        }
+    }
+
+    public void onBlockClickedDefault(World world, BlockPos pos, EntityPlayer player)
+    {
+        super.onBlockClicked(world, pos, player);
+    }
+
+    @Override
+    public final void neighborChanged(IBlockState state, World world, BlockPos pos, Block neighborBlock)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        if (tile != null)
+        {
+            tile.getMicroblockContainer().getPartContainer().onNeighborBlockChange(neighborBlock);
+        }
+        onNeighborBlockChangeDefault(world, pos, state, neighborBlock);
+    }
+
+    public void onNeighborBlockChangeDefault(World world, BlockPos pos, IBlockState state, Block neighborBlock)
+    {
+        super.neighborChanged(state, world, pos, neighborBlock);
+    }
+
+    @Override
+    public final void onNeighborChange(IBlockAccess world, BlockPos pos, BlockPos neighbor)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        if (tile != null)
+        {
+            tile.getMicroblockContainer().getPartContainer().onNeighborTileChange(EnumFacing
+                    .getFacingFromVector(neighbor.getX() - pos.getX(), neighbor.getY() - pos.getY(), neighbor.getZ() - pos.getZ()));
+        }
+        onNeighborChangeDefault(world, pos, neighbor);
+    }
+
+    public void onNeighborChangeDefault(IBlockAccess world, BlockPos pos, BlockPos neighbor)
+    {
+        super.onNeighborChange(world, pos, neighbor);
+    }
+
+    @Override
+    public final void onEntityWalk(World world, BlockPos pos, Entity entity)
+    {
+        onEntityWalkDefault(world, pos, entity);
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        if (tile == null) { return; }
+        tile.getMicroblockContainer().getPartContainer().onEntityStanding(entity);
+    }
+
+    public void onEntityWalkDefault(World worldIn, BlockPos pos, Entity entityIn)
+    {
+        super.onEntityWalk(worldIn, pos, entityIn);
+    }
+
+    @Override
+    public final void onEntityCollidedWithBlock(World world, BlockPos pos, IBlockState state, Entity entity)
+    {
+        onEntityCollidedWithBlockDefault(world, pos, state, entity);
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        if (tile == null) { return; }
+        tile.getMicroblockContainer().getPartContainer().onEntityCollided(entity);
+    }
+
+    public void onEntityCollidedWithBlockDefault(World worldIn, BlockPos pos, IBlockState state, Entity entityIn)
+    {
+        super.onEntityCollidedWithBlock(worldIn, pos, state, entityIn);
+    }
+
+    @Override
+    public final Boolean isAABBInsideMaterial(World world, BlockPos pos, AxisAlignedBB aabb, Material material)
+    {
```

Comment:
Out of curiosity as I don't have the code in front of me, what does null signify and why do we care to have this a ternary return?